projects/plugins/boost/app/modules/cache/Boost_Cache.php

```diff
@@ -254,4 +272,208 @@ public function is_backend() {
 
 		return $is_backend;
 	}
+
+	/*
+	 * Checks if the post type is public.
+	 *
+	 * @param WP_Post $post - The post to check.
+	 * @return bool - True if the post type is public.
+	 */
+	protected function is_visible_post_type( $post ) {
+		$post_type = is_object( $post ) ? get_post_type_object( $post->post_type ) : null;
+		if ( empty( $post_type ) || ! $post_type->public ) {
+			return false;
+		}
+		return true;
+	}
+
+	/*
+	 * Delete the cache for the front page and paged archives.
+	 * This is called when a post is edited, deleted, or published.
+	 *
+	 * @param int $post_id - The id of the post that was edited, deleted, or published.
+	 */
+	protected function delete_front_page_cache( $post ) {
+		$front_page_id = get_option( 'show_on_front' ); // posts page
+		if ( $front_page_id === 'page' ) {
+			$front_page_id = get_option( 'page_on_front' ); // static page
+			if ( $front_page_id === $post->ID ) {
+				$this->delete_cache_for_post( $post->ID );
+			}
+		} else {
+			$this->storage->invalidate_home_page();
+			error_log( 'delete front page cache' ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+		}
+	}
```

Comment:
This is for when you're using a static page for the front page. (See Settings->Reading in wp-admin)

Reply:
Isn't the cache file the one for `/`?

Reply:
No; the site won't always be at `/`; it can be in a subdirectory, for example. 

But that then leaves me wondering how `invalidate_home_page` works, as it deletes the root of the cache directory, which may or may not be the site's homepage.

Reply:
You're both right. While I was sitting in the driver's seat in my car waiting for the roadside assistance yesterday, I tested using a static page for the front page. It was cached as "/", so that function can be simplified.
I'm going to test this PR on my test site in a subdirectory and see how that works and make adjustments.

Reply:
So. Two issues here:
* Site is hosted at /test/ instead of /
* Site is using a static page for the front page and another page for posts

I'll handle the second one first. The first "front_page_id" check above handled that, but it didn't work at all. It did update the static homepage, but never updated the page used to hold the posts.

I've just rewritten that. The static front page is at "/" (or "/test/", it doesn't matter as the post permalink has the right directory) and the posts page in my test example is at /posts/. Both locations are cleared when a post is updated. There might be a case to be made that the static homepage doesn't need to be updated, but some sites will want it updated.

Reply:
I just fixed having a site hosted at /test/ rather than / by passing the path from the home_url() to invalidate_home_page().